pkg/util/tls/tls.go

```diff
@@ -26,21 +28,28 @@ var (
 
 // RegisterFlagsWithPrefix registers flags with prefix.
 func (cfg *ClientConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.BoolVar(&cfg.Enable, prefix+".tls-enable", false, "Enable TLS in the client. This flag is implicitly enabled when any other TLS flag is set. This flag will only be respected by the GRPC client, as for the HTTP client the scheme used in the URL will take precedence.")
 	f.StringVar(&cfg.CertPath, prefix+".tls-cert-path", "", "Path to the client certificate file, which will be used for authenticating with the server. Also requires the key path to be configured.")
 	f.StringVar(&cfg.KeyPath, prefix+".tls-key-path", "", "Path to the key file for the client certificate. Also requires the client certificate to be configured.")
 	f.StringVar(&cfg.CAPath, prefix+".tls-ca-path", "", "Path to the CA certificates file to validate server certificate against. If not set, the host's root CA certificates are used.")
+	f.StringVar(&cfg.ServerName, prefix+".tls-server-name", "", "Override the expected name on the server certificate.")
 	f.BoolVar(&cfg.InsecureSkipVerify, prefix+".tls-insecure-skip-verify", false, "Skip validating server certificate.")
 }
 
 // GetTLSConfig initialises tls.Config from config options
 func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
 	// no tls config given at all
-	if cfg.CertPath == "" && cfg.KeyPath == "" && cfg.CAPath == "" {
+	if cfg.CertPath != "" || cfg.KeyPath != "" || cfg.CAPath != "" || cfg.InsecureSkipVerify || cfg.ServerName != "" {
+		cfg.Enable = true
```

Comment:
We try to never manipulate configs at runtime (wherever possible). In this case I believe it's easy to avoid it, just using a local variable `enabled := cfg.Enable` which can be overridden here.